src/class/SimpleData.ts

```diff
@@ -603,6 +604,36 @@ export default class SimpleData {
         return this
     }
 
+    @logCall()
+    addRank({
+        rankTitle,
```

Comment:
To stay consistent with the other methods, this should probably be named "newKey". You are not creating multiple new keys with this method, so there's no need for a prefix. The user can choose freely how to name the new key.